synapse/handlers/profile.py

```diff
@@ -60,9 +60,14 @@ def __init__(self, hs):
         )
 
         self.user_directory_handler = hs.get_user_directory_handler()
+        self.media_repository = hs.get_media_repository()
 
         self.http_client = hs.get_simple_http_client()
 
+        self.max_avatar_size = hs.config.max_avatar_size
+        self.allowed_avatar_mimetypes = hs.config.allowed_avatar_mimetypes
+        self.media_repo_enabled = hs.config.enable_media_repo
```

Comment:
`enable_media_repo` being false can mean simply that the media repo is on a worker, which would still be fine. I think we should just remove these checks and put a health warning in the config that setting avatars will fail if not using the in built media repo.